src/couch_views/src/couch_views_server.erl

```diff
@@ -0,0 +1,101 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_views_server).
+
+
+-behaviour(gen_server).
+
+
+-export([
+    start_link/0
+]).
+
+
+-export([
+    init/1,
+    terminate/2,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    code_change/3
+]).
+
+
+-define(MAX_WORKERS, 100).
+
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+
+init(_) ->
+    process_flag(trap_exit, true),
+    couch_views_jobs:set_timeout(),
+    State = #{
+        workers => [],
+        num_workers => num_workers()
+    },
+    {ok, spawn_workers(State)}.
+
+
+terminate(_, _St) ->
+    ok.
+
+
+handle_call(Msg, _From, St) ->
+    {stop, {bad_call, Msg}, {bad_call, Msg}, St}.
+
+
+handle_cast(Msg, St) ->
+    {stop, {bad_cast, Msg}, St}.
+
+
+handle_info({'EXIT', Pid, Reason}, State) ->
```

Comment:
If workers was a map like @iilyak suggested could even check membership in the function head :-P

```
handle_info({'EXIT', Pid, Reason}, #{workers := #{Pid := true}}) ->
   {noreply, spawn_workers(...)}

handle_info({'EXIT', Pid, Reason)}, State) ->
    {stop, ...}
```

Reply:
It turns out map key variable matches in the function heads don't work. There is http://erlang.org/doc/man/erlang.html#is_map_key-2 but it's Erlang 21+ only and we are on 19 minimum. It would have to be a case statement then.